uc-crux-llvm/src/UCCrux/LLVM/Run/Explore.hs

```diff
@@ -67,23 +72,24 @@ exploreOne ::
   AppContext ->
   ModuleContext m arch ->
   CruxOptions ->
-  UCCruxLLVMOptions ->
+  LLVMOptions ->
+  ExploreConfig ->
   Crucible.HandleAllocator ->
   FilePath ->
-  String ->
+  DeclSymbol m ->
   IO Stats
-exploreOne appCtx modCtx cruxOpts ucOpts halloc dir func =
+exploreOne appCtx modCtx cruxOpts llOpts exOpts halloc dir declSym =
   do
+    let L.Symbol func = getDeclSymbol declSym
     let logFilePath = dir </> func -<.> ".summary.log"
     logExists <- doesPathExist logFilePath
-    if not logExists || Config.reExplore ucOpts
+    if not logExists || ExConfig.exploreAgain exOpts
```

Comment:
> It would be good to add a comment here explaining why exploration is suppressed when a logfile is present. 

I will add a comment. Just to respond inline, the reasoning is that you might have a large codebase where you've done one "round" of exploration, and you don't want to waste time re-exploring those functions. The logs are just a bit of persistent state that records that this function was already visited.

> Alternatively, since you're already doing async here, you could async create a log file writer with a Control.Concurrent.Chan to these output emitters that would serialize the writes.

I'm not sure I understand what you're suggesting here, can you elaborate?

Reply:
I was confusing this scenario with the one Tristan noted below.  Thanks for clarifying (in comment) the reason for the log file check, and the latter comment could be transplanted to the below discussion (at line 162).

Reply:
Comment added!